benchmarks/graphics_pipeline/GraphicsBenchmarkApp.h

```diff
@@ -164,7 +166,51 @@ class GraphicsBenchmarkApp
         std::string name;
     };
 
+    struct SpherePipelineKey
+    {
+        uint8_t ps;
+        uint8_t vs;
+        uint8_t vertexFormat;
+        uint8_t vertexAttributeLayout;
+        bool    enableDepth;
+        bool    enableAlphaBlend;
+
+        static_assert(kAvailablePsShaders.size() < (1 << (8 * sizeof(ps))));
+        static_assert(kAvailableVsShaders.size() < (1 << (8 * sizeof(vs))));
+        static_assert(kAvailableVbFormats.size() < (1 << (8 * sizeof(vertexFormat))));
+        static_assert(kAvailableVertexAttrLayouts.size() < (1 << (8 * sizeof(vertexAttributeLayout))));
+
+        bool operator==(const SpherePipelineKey& rhs) const
+        {
+            return ps == rhs.ps &&
+                   vs == rhs.vs &&
+                   vertexFormat == rhs.vertexFormat &&
+                   vertexAttributeLayout == rhs.vertexAttributeLayout &&
+                   enableDepth == rhs.enableDepth &&
+                   enableAlphaBlend == rhs.enableAlphaBlend;
+        }
+
+        struct Hash
+        {
+            // Not a good hash function, but good enough.
+            size_t operator()(const SpherePipelineKey& key) const
+            {
+                size_t res = 0;
+
+                res = (res << 2) | key.ps;
```

Comment:
why shift res if it's zero.

Reply:
Keeping it consistent with other field (ps is 2 bit at the moment), compiler will probably remove the (0<<2) anyway.

Reply:
Sad that log2 is not constexpr, otherwise you could do `(res << std::log2(size_ps))` etc

Reply:
`constexpr log2` is not that hard to write.
Although, multiplication works fine, it probably adds a fraction of nano second to the run time which is fine.

Reply:
I believe it would be worthwhile to verify that the number of elements for each option does not exceed the bits allocated in the hash. For example: vs.size() <= (1 << bitsForVs),   ps.size() <= (1 << bitsForPs), so we can avoid collisions and remember to update the hash computation.
Update: I think is was already solved.